src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,92 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private int capacity = DEFAULT_CAPACITY;
+    private T[] list;
+
+    public ArrayList() {
+        list = (T[]) new Object[capacity];
+    }
+
+    public void checkingIndex(int incomingIndex) {
```

Comment:
```suggestion
    private void checkIndex(int incomingIndex) {
```

please make helper methods private and move them below public